### PR TITLE
Lower minimum Python version to 3.7

### LINKED ISSUE #7: Lower minimum Python version to 3.7
I would like to use this as a dependency in a project that has a minimum Python version of 3.7, but this project has a minimum version of 3.8. It doesn't look like there is anything preventing this from running correctly under Python 3.7.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
Fixes #7

### CODE REVIEW

**Review by hrshdhgd — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ readme = "README.md"
 authors = ["cmungall <cjm@berkeleybop.org>"]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.7"
```
[the remark this answers is not in the document]
Comment (pkalita-lbl) [reply pyproject.toml]: Good call. That actually surfaced an issue I didn't catch before which is that this project has a dev dependency on `linkml` so it's minimum version has to be 3.7.6 and not just 3.7.

**Comment by pkalita-lbl** [comment]
Gah! Tests were passing with Python 3.7 locally, but after merging `main` into this branch they no longer pass because of `from importlib import metadata` in `src/prefixmaps/__init__.py`. `importlib.metadata` was added in Python 3.8 🤦🏻 

@hrshdhgd do you know any way to make that import backward compatible?

**Comment by hrshdhgd** [comment]
hmm .... how crucial is supporting python v3.7.x ? I was under the impression we had decided to support everything > v 3.8.0. This may be a Chris question. 

I do not know what we could do for backward compatibility right at the top of my head. I'll start looking.

**Comment by pkalita-lbl** [comment]
I want to use this in the `linkml` project. So either this has to go down to 3.7.6 (the current minimum for `linkml`) or `linkml` has to go up to 3.8.0.

**Comment by hrshdhgd** [comment]
I think the latter option is a more easy solution.

**Comment by cmungall** [comment]
If we like `from importlib import metadata` we should use this universally

does this work for our current release mechanism?

**Comment by hrshdhgd** [comment]
Yes. It gets the `__version__` from one place in the project (the toml file) as opposed to hard-coding it in different places.

**Comment by pkalita-lbl** [comment]
There is a compatibility package (https://github.com/python/importlib_metadata) but it seems like it's getting tripped up on the `src` directory structure.

**Comment by hrshdhgd** [comment]
Oh yeah! for the import we could do a `if else` statement that checks the python version and depending on that, the import statement could be modified. I remember seeing this somewhere.

**Comment by hrshdhgd** [comment]
I meant `try-except` not `if else`
```
try:
    from importlib import metadata
except ImportError: # for Python<3.8
    import importlib_metadata as metadata
__version__ = metadata.version()
```

**Comment by pkalita-lbl** [comment]
Yeah that's what I was attempting to do, but I keep getting this when I run the test suite:

```
ImportError: Failed to import test module: src.prefixmaps
Traceback (most recent call last):
  File "/Users/PAKalita/.pyenv/versions/3.7.13/lib/python3.7/unittest/loader.py", line 470, in _find_test_path
    package = self._get_module_from_name(name)
  File "/Users/PAKalita/.pyenv/versions/3.7.13/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/PAKalita/Work/prefixmaps/src/prefixmaps/__init__.py", line 6, in <module>
    __version__ = version(__name__)
  File "/Users/PAKalita/Work/prefixmaps/.venv/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 585, in version
    return distribution(distribution_name).version
  File "/Users/PAKalita/Work/prefixmaps/.venv/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 558, in distribution
    return Distribution.from_name(distribution_name)
  File "/Users/PAKalita/Work/prefixmaps/.venv/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 215, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: No package metadata was found for src.prefixmaps
```

**Comment by pkalita-lbl** [comment]
Wait, that error was the result of trying to run `poetry run python -m unittest` but I don't get that when trying to run `poetry run pytest`. So maybe that's more of a `unittest` issue than a `importlib_metadata` issue??

**Comment by pkalita-lbl** [comment]
I've added the change to import from `importlib_metadata` as a fallback. Everything seems to be passing now. Any final comments on this?

**Pull request merged**